### PR TITLE
Removes duplicate code in AI functions

### LINKED ISSUE #5454: [AI Clean up] Removes redundancy
### Description

Make a function for `ABILITYEFFECT_WOULD_ABSORB` and `ABILITYEFFECT_ABSORBING` to delete duplicate code from `battle_ai_main.c`

### Discord contact info

_No response_

### PR DESCRIPTION
**Pull request by AlexOn1ine**
`CanAbilityBlockMove`, `CanPartnerAbilityBlockMove` and `CanAbilityAbsorbMove` to remove redundancy in `AI_CheckBadMove` and `IsDamageMoveUnusable`

Not that much code was removed but the important part was to unify both so we don't need to add those abilities to AI functions. We actually didn't have a check for Good As Gold, Wind Rider and we didn't check the opposing partner ability.

Closes #5454

### CODE REVIEW

**Comment by AlexOn1ine** [comment]
Ready for review.

File: src/battle_util.c
Comment (AlexOn1ine) [inline src/battle_util.c:5400]: This part is a bit scuffed but it was worse when I checked them in the function directly. It is cleaner isolated here.

Comment (Pawkkie) [reply src/battle_util.c:5400]: The enums look *chef's kiss* though, I'll try to make time to review tomorrow if no one beats me to it :)

File: include/config/general.h
Comment (Pawkkie) [inline include/config/general.h:9]: What's this removed for?

Comment (AlexOn1ine) [reply include/config/general.h]: reverted

**Review by Pawkkie — DISMISSED** [review]
I've read through this 3 times across a couple of days cuz it's kind of a chunky refactor and I'm trying to be careful, but this looks great to me. Happy to wait for another reviewer's approval as well if you like but I think this looks good to go :)

File: src/battle_main.c
Comment (AlexOn1ine) [inline src/battle_main.c:4204]: note for me. this isn't the right place

**Comment by AlexOn1ine** [comment]
Let's wait with merging until #5489 is merged

**Review by DizzyEggg — CHANGES_REQUESTED** [review]

File: src/battle_util.c
Comment (AlexOn1ine) [inline src/battle_util.c:4161]:
```suggestion
            s32 priority = AI_DATA->aiCalcInProgress ? GetMovePriority(battlerAtk, move) : GetChosenMovePriority(battlerAtk);
```

Comment (AlexOn1ine) [reply src/battle_util.c]: right, good catch

**Comment by AlexOn1ine** [comment]
> Let's wait with merging until #5489 is merged

Ready for re-review. I think it is fine if this one is merged first, Someone will have to take care of conflicts anyways.

**Pull request merged**
